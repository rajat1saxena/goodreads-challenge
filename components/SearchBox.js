/**
 * A search box component for accepting user query.
 */
import React from 'react'

const SearchBox = ({ query, handleQueryChange, handleSearch }) => (
  <div className="container">
    <input
      type="text"
      placeholder="Search GoodReads.com books"
      value={query}
      onChange={ (e) => handleQueryChange(e.target.value) } />
    <button type='submit' onClick={handleSearch}>Search</button>
    <style jsx>{`
          .container {
            display: flex;
            flex-direction: row
          }
          input {
            border: 2px solid #ddd;
            border-radius: 4px;
            padding: 10px 5px;
            flex-grow: 1
          }
        `}
    </style>
  </div>
)

export default SearchBox
