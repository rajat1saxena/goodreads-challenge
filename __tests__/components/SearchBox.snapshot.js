import React from 'react'
import renderer from 'react-test-renderer'
import SearchBox from '../../components/SearchBox.js'

it('Should render correctly', () => {
  const tree = renderer.create(<SearchBox />).toJSON()
  expect(tree).toMatchSnapshot()
})