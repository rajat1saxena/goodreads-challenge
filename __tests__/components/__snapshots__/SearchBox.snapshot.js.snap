// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly 1`] = `
<div
  className="jsx-4012630625 container"
>
  <input
    className="jsx-4012630625"
    onChange={[Function]}
    placeholder="Search GoodReads.com books"
    type="text"
  />
  <button
    className="jsx-4012630625"
    data-testid="submit"
    type="submit"
  >
    Search
  </button>
</div>
`;
